scripts/install-du.ps1

```diff
@@ -94,30 +97,71 @@ function CalcHash {
     [Convert]::ToBase64String($hashAlg.ComputeHash([System.IO.File]::ReadAllBytes($FilePath)))
 }
 
-function WriteExtensionRegistration {
+function WriteContentDownloaderRegistration{
     Param(
-        [Parameter(Mandatory = $true)][string]$BinaryFilePath,
-        [Parameter(Mandatory = $true)][string]$OutputJsonFilePath
+        [Parameter(Mandatory = $true)][string]$BinaryFileNameNoExt,
+        [Parameter(Mandatory = $true)][string]$ExtensionsDir
     )
 
-    $fileSizeInBytes = (Get-Item $BinaryFilePath).Length
-    $fileHash = CalcHash -FilePath $BinaryFilePath
+    $adu_extensions_sources_dir = "$ExtensionsDir/sources"
+    $binaryFilePath = "$adu_extensions_sources_dir/${BinaryFileNameNoExt}.dll"
+
+    $fileSizeInBytes = (Get-Item $binaryFilePath).Length
+    $fileHash = CalcHash -FilePath $binaryFilePath
 
     $registration = [ordered]@{
-        fileName="$BinaryFilePath"
-        sizeInBytes=$FileSizeInBytes
+        fileName="$binaryFilePath"
+        sizeInBytes=$fileSizeInBytes
         hashes = @{
             sha256="$fileHash"
         }
     }
 
-    $registration | ConvertTo-Json | Set-Content $OutputJsonFilePath
+    $registration | ConvertTo-Json | Set-Content "$ExtensionsDir/content_downloader/extensions.json"
+}
+
+function WriteContentHandlerRegistration{
+    Param(
+        [Parameter(Mandatory = $true)][string]$BinaryFileNameNoExt,
+        [Parameter(Mandatory = $true)][string]$ExtensionName,
+        [Parameter(Mandatory = $true)][string]$ExtensionsDir
+    )
+
+    $adu_extensions_sources_dir = "$ExtensionsDir/sources"
+
+    $binaryFilePath = "$adu_extensions_sources_dir/${BinaryFileNameNoExt}.dll"
+    $fileSizeInBytes = (Get-Item $binaryFilePath).Length
+    $fileHash = CalcHash -FilePath $binaryFilePath
+
+    $numUnderscore = ($ExtensionName | Select-String -Pattern '_' -AllMatches).Matches.Count
+    $handlerId = $ExtensionName
+    if ($numUnderscore -eq 2) {
+        $handlerId = $ExtensionName -replace '(.+?)_(.+?)_(.+)', '$1/${2}:$3'
+    }
+    elseif ($numUnderscore -eq 1) {
+        $handlerId = $ExtensionName -replace '(.+?)_(.+)', '$1/$2'
+    }
+    else {
+        Show-Warning -Message "Expected 1 or 2 underscores in content handler name '$ExtensionName'"
+    }
+
+    $registration = [ordered]@{
+        fileName="$binaryFilePath"
+        sizeInBytes=$fileSizeInBytes
+        hashes = @{
+            sha256="$fileHash"
+        }
+        handlerId = "$handlerId"
+    }
+
+    $registration | ConvertTo-Json | Set-Content "$ExtensionsDir/update_content_handlers/$ExtensionName/content_handler.json"
```

Comment:
Set-Content -> Out-File ? (unclear what the diff is)

Reply:
Found this:
https://stackoverflow.com/questions/10655788/powershell-set-content-and-out-file-what-is-the-difference

Yes, switching to Out-File. Set-Content can do some things that Out-File can't but unneeded here.